tsnet/tsnet.go

```diff
@@ -517,6 +541,27 @@ func (s *Server) getAuthKey() string {
 	return os.Getenv("TS_AUTH_KEY")
 }
 
+func (s *Server) getClientSecret() string {
+	if v := s.ClientSecret; v != "" {
+		return v
+	}
+	return os.Getenv("TS_CLIENT_SECRET")
+}
+
+func (s *Server) getClientID() string {
+	if v := s.ClientID; v != "" {
+		return v
+	}
+	return os.Getenv("TS_CLIENT_ID")
+}
+
+func (s *Server) getIDToken() string {
+	if v := s.IDToken; v != "" {
+		return v
+	}
+	return os.Getenv("TS_ID_TOKEN")
+}
+
```

Comment:
Naming: I have tried to conform with the parameter names for the `tailscale up` command (see [here](https://github.com/tailscale/tailscale/blob/ece6e27f39ceb11b4c51ef4bfd317cacb5203d89/cmd/tailscale/cli/up.go#L100-L102)), adding a `TS_` prefix in the case of the environment variables. I'd be happy to hear other ideas though!